packages/core/src/error/error.factory.ts

```diff
@@ -0,0 +1,34 @@
+import chalk from 'chalk';
+import { CoreError } from './error.model';
+
+export type CoreErrorOptions = {
+  contextMethod: string;
+  contextPackage?: string;
+};
+
+export const coreErrorStackTraceFactory = (opts: CoreErrorOptions) => (message: string, stack: NodeJS.CallSite[]) => {
+  const [method, file] = stack;
+  const [line, col] = [file.getLineNumber() || 0, file.getColumnNumber() || 0];
+  const packageName = opts.contextPackage || '@marblejs/core';
+  const methodName = opts.contextMethod + ' : ' + (method.getMethodName() || '-');
+  const fileName = file.getFileName() || '';
+
+  return `
+    ${chalk.red(`${packageName} error:`)}
+
+      🚨  ${message}
+
+      👉  ${chalk.yellow.bold(methodName)}
+        @ file: ${chalk.underline(fileName)}
+        @ line: [${line.toString()}:${col.toString()}]
```

Comment:
I think that the whole stack *must* be included here, not only the last file. I'd append a series of something like:
```ts
stackTraceElements.map(e => `@ ${e}`)
```

Reply:
Are you sure? IMO the rest of the stack trace is not important because it will list all node internals which are rubbish.

Reply:
Can you trigger the error in a sample project and check the stack trace? It should lead from the error to developer's code, i.e. to the endpoint he defined incorrectly.

Reply:
```
/Users/jflakus/Developer/home/@marblejs-marble/packages/core/dist/effects/effects.factory.js:12
            throw error_factory_1.coreErrorFactory('Path cannot be empty', coreErrorOptions);
                                  ^
CoreError: Path cannot be empty
    at Object.EffectFactory.matchPath (/Users/jflakus/Developer/home/@marblejs-marble/packages/core/dist/effects/effects.factory.js:12:35)
    at Object.<anonymous> (/Users/jflakus/Developer/home/@marblejs-marble/example/src/controllers/api.controller.ts:30:15)
    at Module._compile (module.js:635:30)
    at Module.m._compile (/Users/jflakus/Developer/home/@marblejs-marble/node_modules/ts-node/src/index.ts:439:23)
    at Module._extensions..js (module.js:646:10)
    at Object.require.extensions.(anonymous function) [as .ts] (/Users/jflakus/Developer/home/@marblejs-marble/node_modules/ts-node/src/index.ts:442:12)
    at Module.load (module.js:554:32)
    at tryModuleLoad (module.js:497:12)
    at Function.Module._load (module.js:489:3)
    at Module.require (module.js:579:17)
```

the output from `ts-node` compiler

Reply:
As I said - it is a rubbish

Reply:
The first record in stack trace is removed using `Error.captureStackTrace` method. It is useful for hiding implementation details of error generation from an end user.